flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/ReconciliationUtils.java

```diff
@@ -305,4 +307,46 @@ public static boolean isJobRunning(FlinkDeploymentStatus status) {
                         .name()
                         .equals(status.getJobStatus().getState());
     }
+
+    /**
+     * In case of validation errors we need to (temporarily) reset the old spec so that we can
+     * reconcile other outstanding changes, instead of simply blocking.
+     *
+     * <p>This is only possible if we have a previously reconciled spec.
+     *
+     * <p>For in-flight application upgrades we need extra logic to set the desired job state to
+     * running
+     *
+     * @param deployment The current deployment to be reconciled
+     * @param validationError Validation error encountered for the current spec
+     * @return True if the spec was reset and reconciliation can continue. False if nothing to
+     *     reconcile.
+     */
+    public static <
+                    SPEC extends AbstractFlinkSpec,
+                    STATUS extends CommonStatus<SPEC>,
+                    CR extends CustomResource<SPEC, STATUS>>
+            boolean applyValidationErrorAndResetSpec(CR deployment, String validationError) {
```

Comment:
Hi @gyfora, I have another concern here. For the sessionJob, if the validationError comes from the change by the session cluster spec, this solution may not work, Right ?

Reply:
Could you give me an example where this would not work?